ucsb-cs156-f24/proj-dining-f24-15
# MenuItemReview post (all users)

Closes #7
dokku link: https://proj-dining-sophiaray2101.dokku-15.cs.ucsb.edu/

- [x] Created the controller and test files for post
- [x] 100% jacoco test coverage
- [x] 100% mutation coverage
- [x] works on both localhost and dokku

What to test:
1. Make sure there are no internal server errors trying to post
2. Make sure that if you are not logged in you cannot post
3. Check that there is a 404 error when trying to post to a non existent id.

When trying to do a post, I would first do the GET on the menu item to ensure that you know which item id you are leaving a review for and that the id's match

*NEW CHANGES*
1. there were too many issues with the get all and it wouldnt work no matter what I did, so I am just focusing on the post
2. CreatedDate and LastEditedDate are now automatically set as the local time at that moment. in the future when there are edits to be made, the LastEditedDate can be updated in that controller method.
3. StudentUserId is grabbed from the logged in user automatically.
5. Status is initally assigned as "Awaiting Moderation".
6. Moderator comments and userId are both not set.
7. Added ratings and reviewText to the entity and all other necessary files

<img width="1170" alt="Screenshot 2024-11-25 at 9 37 45 PM" src="https://github.com/user-attachments/assets/6699d42f-dd33-4952-8ccb-d1868de1e94d">
